sdk/client.go

```diff
@@ -0,0 +1,120 @@
+package sdk
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	healthcheck "github.com/ONSdigital/dp-api-clients-go/v2/health"
+	"github.com/ONSdigital/dp-feedback-api/models"
+	sdkError "github.com/ONSdigital/dp-feedback-api/sdk/errors"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+)
+
+// package level constants
+const (
+	Service          = "dp-feedback-api"
+	FeedbackEndpoint = "%s/v1/feedback"
+	Authorization    = "Authorization"
+	BearerPrefix     = "Bearer "
+)
+
+// HTTPClient is the interface that defines a client for making HTTP requests
+type HTTPClient interface {
+	Do(ctx context.Context, req *http.Request) (*http.Response, error)
+}
+
+// APIClient implementation of permissions.Store that gets permission data from the permissions API
+type Client struct {
+	hcCli *healthcheck.Client
+}
+
+// Options is a struct containing for customised options for the API client
+type Options struct {
+	AuthToken string
```

Comment:
I'm not sure about this.  I mean it's nice to have functions for auto-complete reasons for the Options, but I think there's more headers required ( i.e. maybe collections ), so maybe this is more hassle that just allowing folks to set the headers.

Reply:
thanks for the feedback! 
Yes, I wsan't 100% sure as well, I thought it makes sense to simplify the usage, but it's at the expense of flexibility. I can change it in the future if required.